sdk/src/main/java/com/okta/spring/boot/sdk/OktaSdkConfig.java

```diff
@@ -108,19 +111,40 @@ protected ClientCredentials oktaSdkClientCredentials() {
     }
 
     /**
-     * Spring Boot conditional based on the existance of the {code}okta.client.token{code} property.
+     * Spring Boot conditional based on the existence of the properties:
+     * <pre>{@code
+     *  - okta.client.token
+     *  - okta.client.orgUrl
+     * }</pre>
+     *
+     * <p> If <code>'okta.client.orgUrl'</code> is absent, try to resolve it from <code>'okta.oauth2.issuer'</code>.
      */
-    static class OktaApiTokenCondition extends SpringBootCondition {
+    static class OktaApiConditions extends SpringBootCondition {
+
+        private static final Pattern ORG_URL_PATTERN = Pattern.compile("^(?<orgUrl>http(s)?://[^/]+)/?.*$");
 
         @Override
         public ConditionOutcome getMatchOutcome(ConditionContext context, AnnotatedTypeMetadata metadata) {
+            ConditionMessage.Builder message = ConditionMessage.forCondition("Okta Api Condition");
 
-            ConditionMessage.Builder message = ConditionMessage.forCondition("Okta Api Token Condition");
             String tokenValue = context.getEnvironment().getProperty("okta.client.token");
-            if (StringUtils.hasText(tokenValue)) {
-                return ConditionOutcome.match(message.foundExactly("provided API token"));
+            if (!StringUtils.hasText(tokenValue)) {
+                return ConditionOutcome.noMatch(message.didNotFind("provided API token").atAll());
             }
-            return ConditionOutcome.noMatch(message.didNotFind("provided API token").atAll());
+
+            String orgUrlValue = context.getEnvironment().getProperty("okta.client.orgUrl");
+            if (!StringUtils.hasText(orgUrlValue)) {
+                String issuerValue = context.getEnvironment().getProperty("okta.oauth2.issuer");
+                if (!StringUtils.hasText(issuerValue)) {
+                    return ConditionOutcome.noMatch(message.didNotFind("provided API orgUrl").atAll());
+                }
+                Matcher matcher = ORG_URL_PATTERN.matcher(issuerValue);
+                if (matcher.find()) {
+                    System.setProperty("okta.client.orgUrl", matcher.group("orgUrl"));
```

Comment:
System properties should not be set here. This could potentially override configuration from other mechanisms leading to hard to find bugs down the road.

All of our property aliasings happen in the post env processors. (this `Condition` shouldn't mutate state whenever possible)

Reply:
Done